src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,168 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FaChevronCircleLeft } from 'react-icons/fa';
+import Layer from '../../layers/Layer';
+import img from '../../images/baselayer/osm.baselayer.png';
+
+import './BaseLayerSwitcher.scss';
+
+const propTypes = {
+  /**
+   * An array of react-spatial layers.
+   */
+  layers: PropTypes.arrayOf(PropTypes.instanceOf(Layer)).isRequired,
+
+  /**
+   * Object containing relative paths to the base layer images.
+   */
+  layerImages: PropTypes.objectOf(PropTypes.string),
+
+  /**
+   * CSS class to apply on the container.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Button titles.
+   */
+  titles: PropTypes.shape({
+    button: PropTypes.string,
+    openSwitcher: PropTypes.string,
+    closeSwitcher: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  className: 'rs-base-layer-switcher',
+  titles: {
+    button: 'Base layers',
+    openSwitcher: 'Open Baselayer-Switcher',
+    closeSwitcher: 'Close Baselayer-Switcher',
+  },
+  layerImages: undefined,
+};
+
+const getNextImage = (currentLayer, layers, layerImages) => {
+  const currentIndex = layers.indexOf(
+    layers.find(layer => layer === currentLayer),
+  );
+  const nextIndex = currentIndex + 1 === layers.length ? 0 : currentIndex + 1;
+  return layerImages.filter(layer => layer)[
+    Object.keys(layerImages)[nextIndex]
+  ];
+};
+
+let timeout;
+
+function BaseLayerSwitcher({ layers, layerImages, className, titles }) {
+  const baseLayers = layers.filter(layer => layer.getIsBaseLayer());
+  const [switcherOpen, setSwitcherOpen] = useState(false);
+  const [isClosed, setIsClosed] = useState(true);
+  const [currentLayer, setCurrentlayer] = useState(
+    baseLayers.find(layer => layer.getVisible()),
+  );
+  /* Images are loaded from props if provided, fallback from layer */
+  const images = layerImages
+    ? Object.keys(layerImages).map(layerImage => layerImages[layerImage])
+    : baseLayers.map(layer => layer.previewImage);
+
+  if (!baseLayers || baseLayers.length < 2) {
+    return null;
+  }
+
+  useEffect(() => {
+    /* Used for correct layer image render with animation */
+    if (!switcherOpen) {
+      window.clearTimeout(timeout);
+      timeout = window.setTimeout(() => {
+        setIsClosed(true);
+      }, 200);
+      return;
+    }
+    setIsClosed(false);
+  }, [switcherOpen]);
```

Comment:
this useEffect should return a function that clears also  the timeout

Reply:
did you mean like this?

Reply:
No see the doc https://de.reactjs.org/docs/hooks-effect.html#effects-with-cleanup

Reply:
Second try:
`useEffect(() => {
    /* Used for correct layer image render with animation */
    if (!switcherOpen) {
      const timeout = setTimeout(() => {
        setIsClosed(true);
      }, 200);
      return () => clearTimeout(timeout);
    }
    return setIsClosed(false);
  }, [switcherOpen]);`

Reply:
You were close :
```
useEffect(() => {
    /* Used for correct layer image render with animation */
   let timeout;
    if (!switcherOpen) {
      timeout = setTimeout(() => {
        setIsClosed(true);
      }, 200);
      } else {
        setIsClosed(false);
     }
      return () => clearTimeout(timeout);
    }
    return setIsClosed(false);
    return () => clearTimeout(timeout);
  }, [switcherOpen]);

```

Reply:
Thanks